### PR TITLE
lintr support

### PR DESCRIPTION
**Pull request by krlmlr**
Still a bit work to do. It should be possible to resolve the unmasked issues and enable
all except of `single_quotes_linter` and `camel_case_linter` with reasonable effort.

See https://github.com/krlmlr/plyr/blob/lint/lint.md for a summary.

### CODE REVIEW

**Comment by hadley** [comment]
I'm not interested in making any major changes to plyr at this point - I don't think there's a lot of value in linting the existing code.

**Comment by krlmlr** [comment]
- Existing code will be slightly more readable
- This is an investment that will increase code quality for future patches while at the same time reducing review effort. (Style nitpicks will be handled automatically.)

**Comment by hadley** [comment]
I really don't imagine any more major releases of plyr. But if you want to invest the time doing this, I'll review the PR.

**Comment by krlmlr** [comment]
Ready for review.

File: R/idataframe.r [hunk 1]
```diff
@@ -27,7 +27,8 @@ idata.frame <- function(df) {
       } else {
         stop("Immutable")
       }
-    }, list(name = name)), envir=self)
+    },
+    list(name = name)), envir=self)
```
Comment (hadley) [inline R/idataframe.r]: Shouldn't the second equals have spaces around it?

**Comment by hadley** [comment]
I'm not in love with the need for `#nolint`, but otherwise it's fine.

File: R/idataframe.r (hunk 1 above)
Comment (krlmlr) [inline R/idataframe.r]: Probably yes, but lintr doesn't check this currently. Can we leave this for now?

Comment (hadley) [inline R/idataframe.r]: Yeah, it's not a big deal

**Pull request merged**